src/main/java/io/cdap/e2e/pages/locators/CdfConnectionLocators.java

```diff
@@ -50,4 +51,20 @@ public static By locatorOfDataTableTitle(String title) {
   public static By locatorOfPipelineTypeButton(String pipelineType) {
     return By.xpath("//a//*[text()='" + pipelineType + "']");
   }
+
+  public static WebElement locateConnectionType(String connectionType) {
+    return SeleniumDriver.getDriver().findElement(
+      By.xpath("//div[@data-cy='categorized-connection-type-" + connectionType + "']"));
+  }
+
+  public static WebElement locateConnectionActionMenu(String connectionName) {
+    return SeleniumDriver.getDriver().findElement(
+      By.xpath("//*[text()='" + connectionName + "']/parent::a/following-sibling::div"));
```

Comment:
can this locator be improved so that it is more specific? Currently, according to my understanding, it will select the `div` following `connectionName` on the dom.

Reply:
Changed locator